src/api.ts

```diff
@@ -79,6 +79,9 @@ export interface FernsRouterOptions<T> {
     user?: User,
     query?: Record<string, any>
   ) => Record<string, any> | null | Promise<Record<string, any> | null>;
+  /** queryTransform is a function to transform the query params before they are used to query the database. This can be
+   * used to run intermediate queries to generate query fields for the main query. Runs before the queryFilter.  */
+  queryTransform?: (query: Record<string, any>) => Record<string, any>;
```

Comment:
Can you write a test? And what would you do here that you wouldn't do in query filter?